### PR TITLE
Fixes #16  - do not crash on exception in plugins registration code

### LINKED ISSUE #16: Throwing plugins crash the entire generator
`ManagedWrapperProjectsPlugin` and `NativeProjectsPlugin` need `TfsProjRoot`, which is not needed for other plugins. It's a bit clumsy that users of these other plugins need to manually remove the offending ones.

Another plugin requires Visual Studio to be installed, which some of the people in my team don't have and shouldn't need

### PR DESCRIPTION
**Pull request by FreeApophis**
* add verbose option to show stack traces
* add ExcepionFormatter Service
* add GlobalOptions
* Fix Typo

### CODE REVIEW

File: MetaGenerator/ExceptionFormatter.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        public ExceptionFormatter(IConsoleWriter consoleWriter, IGlobalOptions globalOptions, IExecutingPluginAssemblyProvider executingPluginAssemblyProvider)
+        {
+            _consoleWriter = consoleWriter;
+            _globalOptions = globalOptions;
+            _executingPluginAssemblyProvider = executingPluginAssemblyProvider;
+        }
+
+        public void FormatException(Exception exception)
+        {
+            _consoleWriter.WriteLine($"Exception in plugin: {Plugin()}");
```
Comment (tautropfli) [inline MetaGenerator/ExceptionFormatter.cs]: When there is no plugin, this reads as "Exception in plugin: MetaGenerator.exe". That's odd.

File: MetaGenerator/ExceptionFormatter.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            _consoleWriter.WriteLine(exception.Message.Pastel(Color.OrangeRed));
+
+            if (_globalOptions.Verbose && exception.StackTrace != null)
```
Comment (tautropfli) [inline MetaGenerator/ExceptionFormatter.cs]:
```suggestion
            if (_globalOptions.Verbose && exception.StackTrace is { })
```

**Review by tautropfli — APPROVED** [review]

**Review by tautropfli — DISMISSED** [review]
Accidentally approved

File: MetaGenerator/ExceptionFormatter.cs (hunk 1 above)
Comment (FreeApophis) [reply MetaGenerator/ExceptionFormatter.cs]: Changed to module according to our discussion

**Review by dngroth — APPROVED** [review]

**Pull request merged**